apps/builder/app/builder/features/breakpoints/cascade-indicator.tsx

```diff
@@ -0,0 +1,126 @@
+import type { Breakpoint } from "@webstudio-is/project-build";
+import { isBaseBreakpoint } from "~/builder/shared/breakpoints";
+
+const cascadeIndicatorStyle = css({
+  position: "absolute",
+  bottom: 0,
+  height: 2,
+  transition: "150ms width, 150ms left, 150ms right",
+  '&[data-direction="left"]': {
+    background: theme.colors.backgroundGradientHorizontal,
+  },
+  '&[data-direction="right"]': {
+    background: theme.colors.backgroundGradientHorizontalReverse,
+  },
+});
+
+const calcIndicatorStyle = ({
+  buttonLeft,
+  buttonWidth,
+  containerWidth,
+  selectedBreakpoint,
+}: {
+  buttonLeft: number;
+  buttonWidth: number;
+  containerWidth: number;
+  selectedBreakpoint: Breakpoint;
+}) => {
+  if (isBaseBreakpoint(selectedBreakpoint)) {
+    return {
+      left: {
+        left: 0,
+        width: buttonLeft + buttonWidth / 2,
+      },
+      right: {
+        width: containerWidth - buttonLeft - buttonWidth / 2,
+        left: buttonLeft + buttonWidth / 2,
+      },
+    };
+  }
+  if (selectedBreakpoint.minWidth !== undefined) {
+    return {
+      left: {
+        right: buttonLeft + buttonWidth,
+        width: buttonLeft + buttonWidth,
+        left: 0,
+      },
+      right: {
+        left: 0,
+        width: 0,
+      },
+    };
+  }
+
+  return {
+    left: {
+      left: 0,
+      width: 0,
+    },
+    right: {
+      width: containerWidth - buttonLeft,
+      left: buttonLeft,
+    },
+  };
+};
+
+const useSizes = ({
+  buttonRef,
+  selectedBreakpoint,
+}: {
+  buttonRef: RefObject<HTMLButtonElement | null>;
+  selectedBreakpoint?: Breakpoint;
+}) => {
+  const [buttonLeft, setButtonLeft] = useState<number>();
+  const [buttonWidth, setButtonWidth] = useState<number>();
+  const [containerWidth, setContainerWidth] = useState<number>();
+
+  useEffect(() => {
+    if (buttonRef.current) {
+      setButtonLeft(buttonRef.current.offsetLeft);
+      setButtonWidth(buttonRef.current.offsetWidth);
+      setContainerWidth(buttonRef.current.parentElement?.offsetWidth);
+    }
+  }, [selectedBreakpoint, buttonRef]);
+
+  if (
+    buttonLeft === undefined ||
+    containerWidth === undefined ||
+    buttonWidth === undefined
+  ) {
+    return;
+  }
+  return { buttonLeft, containerWidth, buttonWidth };
+};
+
+export const CascadeIndicator = ({
+  buttonRef,
+}: {
+  buttonRef: RefObject<HTMLButtonElement | null>;
+}) => {
+  const selectedBreakpoint = useStore(selectedBreakpointStore);
+  const sizes = useSizes({ buttonRef, selectedBreakpoint });
+  if (selectedBreakpoint === undefined || sizes === undefined) {
+    return null;
+  }
+
+  const indicatorStyle = calcIndicatorStyle({ ...sizes, selectedBreakpoint });
+
+  return (
+    <>
+      <div
+        className={cascadeIndicatorStyle()}
+        data-direction="left"
+        style={indicatorStyle.left}
+      />
+      <div
+        className={cascadeIndicatorStyle()}
+        data-direction="right"
```

Comment:
In UI I see single indicator, what the reason to have left and right?
Any comments why we need both would be good.